### PR TITLE
Extracted Core components to a seperate library based on .NET Standard 2.0 and upgraded to .NET Core 3.1 (LTS)

### PR DESCRIPTION
**Pull request by rGunti**
This pull request contains the following changes:
- The executable project was updated to .NET Core 3.1 since .NET Core 2.0 is long out of support.
- Extracted the core components (basically everything except the `Program.cs`) to a separate library suited for publishing via NuGet.

**Motivation for Pull Request:**
I am the author of a small GUI tool called [BeatSaberKeeper](https://github.com/rGunti/BeatSaberKeeper) which recently switched to DepotDownloader for downloading older versions of the VR game "[Beat Saber](https://www.beatsaber.com/)" and creating backup archives of modded game installations. Currently, I am calling the executable of DepotDownloader to run download processes but this is very user-unfriendly so I'd like to use the power of DepotDownloader natively.

### CODE REVIEW

File: DepotDownloader.Core/Steam3Session.cs [hunk 1]
```diff
@@ -1,700 +1,700 @@
...
+            }
+
+            Console.WriteLine( "Got {0} licenses for account!", licenseList.LicenseList.Count );
+            Licenses = licenseList.LicenseList;
+
+            foreach ( var license in licenseList.LicenseList )
+            {
+                if ( license.AccessToken > 0 )
+                {
+                    PackageTokens.Add(license.PackageID, license.AccessToken);
```
Comment (xPaw) [inline DepotDownloader.Core/Steam3Session.cs:652]: Don't change this, as it will crash on duplicates.

**Comment by xPaw** [comment]
Your PR shows entire files as changed but that's just line endings, can you preserve the original ones?

**Comment by rGunti** [comment]
Sure! Will check on what I can do.

**Comment by rGunti** [comment]
I have trouble getting the line endings back. Can you maybe share your preferences so I can adjust my environment (or ideally add a `.gitattributes` so this doesn't happen again)?

File: DepotDownloader.Core/DepotDownloader.Core.csproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFramework>netstandard2.1</TargetFramework>
```
Comment (rGunti) [inline DepotDownloader.Core/DepotDownloader.Core.csproj:4]: `.TryAdd()` doesn't exist in .NET Standard 2.0 (for some bizarre reason)

**Comment by rGunti** [comment]
I will close this PR and reopen a new one once I fixed the merge conflicts.

**Pull request closed without merging**